vhdbuilder/scripts/automate_release.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+set -euo pipefail
+source vhdbuilder/scripts/automate_helpers.sh
+set +x
+
+EV2_ARTIFACT_PIPELINE_ID="319030" 
+SIG_RELEASE_PIPELINE_ID="494"
+
+RELEASE_EV2_ARTIFACTS_ALIAS_NAME="ev2_artifacts"
+RELEASE_API_ENDPOINT="https://msazure.vsrm.visualstudio.com/CloudNativeCompute/_apis/Release/releases?api-version=5.1"
+
+ADO_PAT="${ADO_PAT:-""}"
+VHD_BUILD_ID="${VHD_BUILD_ID:-""}"
+
+trigger_ev2_artifacts() {
+    echo "creating EV2 artifacts for VHD build with ID: $VHD_BUILD_ID"
+
+    RESPONSE=$(az pipelines run --id $EV2_ARTIFACT_PIPELINE_ID --variables "VHD_PIPELINE_RUN_ID=$VHD_BUILD_ID")
+    EV2_BUILD_ID=$(echo "$RESPONSE" | jq -r '.id')
+    EV2_BUILD_NUMBER=$(echo "$RESPONSE" | jq -r '.buildNumber')
+    STATUS="$(az pipelines runs show --id $EV2_BUILD_ID | jq -r '.status')"
+
+    while [ "${STATUS,,}" == "notstarted" ] || [ "${STATUS,,}" == "inprogress" ]; do
+        echo "EV2 artifact build $EV2_BUILD_ID is still in-progress..."
+        sleep 60
+        STATUS="$(az pipelines runs show --id $EV2_BUILD_ID | jq -r '.status')"
+    done
+
+    if [ "${STATUS,,}" != "completed" ]; then
```

Comment:
status is completed even for failed builds:

eg https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=94897542&view=results

➜  ~ az pipelines runs show --id 94897542 | jq -r '.status'
completed

➜  ~ az pipelines runs show --id 94897542 | jq -r '.result'
failed

so we'd end up creating SIG releases even for failed ev2 artifact builds. although rare, I think we should still change this evaluation condition to avoid getting in this state ever.

something like:

if status != completed or result == failed
  exit with error
fi

Reply:
lol I completely missed that

Reply:
fixed